server/src/main/java/org/opensearch/gateway/PrimaryShardAllocator.java

```diff
@@ -484,12 +530,93 @@ static class NodesToAllocate {
      * by the allocator for allocating to the node that holds the shard copy.
      */
     private static class DecidedNode {
-        final NodeGatewayStartedShards nodeShardState;
+        final NodeShardState nodeShardState;
         final Decision decision;
 
-        private DecidedNode(NodeGatewayStartedShards nodeShardState, Decision decision) {
+        private DecidedNode(NodeShardState nodeShardState, Decision decision) {
             this.nodeShardState = nodeShardState;
             this.decision = decision;
         }
     }
+
+    /**
+     * The NodeShardState class represents the state of a node shard in a distributed system.
+     * It includes several key data points about the shard state, such as its allocation ID,
+     * whether it's a primary shard, any store exception, the replication checkpoint, and the
+     * DiscoveryNode it belongs to.
+     */
+    protected static class NodeShardState {
```

Comment:
As discussed in these comments https://github.com/opensearch-project/OpenSearch/pull/8916/files#r1418498209 and https://github.com/opensearch-project/OpenSearch/pull/8916/files#r1418502178 , we can get rid of this class .

Reply:
Removed the class as recommended